agent/agent-profiler/agent-profiler-api/gradle.lockfile

```diff
@@ -1,7 +1,7 @@
 # This is a Gradle generated file for dependency locking.
 # Manual edits can break the build and are not advised.
 # This file is expected to be part of source control.
-com.azure:azure-core:1.17.0=runtimeClasspath
+com.azure:azure-core:1.18.0=runtimeClasspath
```

Comment:
we don't need to update Statsbeat instrumentation for azure sdk, right?  It's always going to stay as "Azure-OpenTelemetry"?  Just want to double check.  We can remove the comment for this line:

`INSTRUMENTATION_MAP.put("Azure-OpenTelemetry", 6); // bridged by azure-core-1.14 module`

Reply:
> `INSTRUMENTATION_MAP.put("Azure-OpenTelemetry", 6); // bridged by azure-core-1.14 module`

this comment is related to our instrumentation of azure-core library (when applications we are monitoring use it)

this PR is updating our internal usage of azure-core library (which we now use instead of Apache HttpClient)